src/operator/sparse_reg-inl.h

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file sparse_reg-inl.h
+ * \brief
+*/
+#ifndef MXNET_OPERATOR_SPARSE_REG_INL_H_
+#define MXNET_OPERATOR_SPARSE_REG_INL_H_
+#include <dmlc/logging.h>
+#include <mxnet/operator.h>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+#include "./mshadow_op.h"
+#include "./operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+namespace sparsereg {
+enum SparseRegOpInputs {kData};
+enum SparseRegOpOutputs {kOut};
+enum SparseRegOpAuxiliary {kMovingAvg};
+enum SparseRegBackResource {kTempSpace};
+}  // namespace sparsereg
+
+struct SparseRegParam : public dmlc::Parameter<SparseRegParam> {
+  float penalty;
+  float sparseness_target;
+  float momentum;
+  DMLC_DECLARE_PARAMETER(SparseRegParam) {
+    DMLC_DECLARE_FIELD(sparseness_target).set_default(0.1)
+    .set_range(0, 1)
+    .describe("The sparseness target");
+    DMLC_DECLARE_FIELD(penalty).set_default(0.001)
+    .describe("The tradeoff parameter for the sparseness penalty");
+    DMLC_DECLARE_FIELD(momentum).set_default(0.9)
+    .set_range(0, 1)
+    .describe("The momentum for running average");
+  }
+};  // struct SparseRegParam
+
+template<typename xpu>
+class SparseRegOp : public Operator {

```

Comment:
This is like another special operator class. 
- Backward is not the gradient on output wrt. input.

Currently only XXXOutput operators have this kind of behavior for output loss functions. If we are going to add this as special class, think about naming convention. Or add it as a true operator, output regualrization value and backprop the gradient